x-pack/plugins/ml/server/saved_objects/checks.ts

```diff
@@ -0,0 +1,298 @@
+          namespaces,
+          checks: {
+            jobExists,
+            datafeedExists,
+          },
+        };
+      }
+    );
+
+    const nonSpaceSavedObjects = await _loadAllJobSavedObjects();
+    const nonSpaceADObjectIds = new Set(
+      nonSpaceSavedObjects
+        .filter(({ type }) => type === 'anomaly-detector')
+        .map(({ jobId }) => jobId)
+    );
+    const nonSpaceDFAObjectIds = new Set(
+      nonSpaceSavedObjects
+        .filter(({ type }) => type === 'data-frame-analytics')
+        .map(({ jobId }) => jobId)
+    );
+
+    const adObjectIds = new Set(
+      savedObjectsStatus.filter(({ type }) => type === 'anomaly-detector').map(({ jobId }) => jobId)
+    );
+    const dfaObjectIds = new Set(
+      savedObjectsStatus
+        .filter(({ type }) => type === 'data-frame-analytics')
+        .map(({ jobId }) => jobId)
+    );
+
+    const anomalyDetectors = adJobs.jobs
+      .filter(({ job_id: jobId }) => {
+        // only list jobs which are in the current space (adObjectIds)
+        // or are not in any spaces (nonSpaceADObjectIds)
+        return adObjectIds.has(jobId) === true || nonSpaceADObjectIds.has(jobId) === false;
+      })
+      .map(({ job_id: jobId }) => {
+        const datafeedId = datafeeds.datafeeds.find((df) => df.job_id === jobId)?.datafeed_id;
+        return {
+          jobId,
+          datafeedId: datafeedId ?? null,
+          checks: {
+            savedObjectExits: nonSpaceADObjectIds.has(jobId),
+          },
+        };
+      });
+
+    const dataFrameAnalytics = dfaJobs.data_frame_analytics
+      .filter(({ id: jobId }) => {
+        // only list jobs which are in the current space (dfaObjectIds)
+        // or are not in any spaces (nonSpaceDFAObjectIds)
+        return dfaObjectIds.has(jobId) === true || nonSpaceDFAObjectIds.has(jobId) === false;
+      })
+      .map(({ id: jobId }) => {
+        return {
+          jobId,
+          datafeedId: null,
+          checks: {
+            savedObjectExits: nonSpaceDFAObjectIds.has(jobId),
+          },
+        };
+      });
+
+    return {
+      savedObjects: {
+        'anomaly-detector': savedObjectsStatus.filter(({ type }) => type === 'anomaly-detector'),
+        'data-frame-analytics': savedObjectsStatus.filter(
+          ({ type }) => type === 'data-frame-analytics'
+        ),
+      },
+      jobs: {
+        'anomaly-detector': anomalyDetectors,
+        'data-frame-analytics': dataFrameAnalytics,
+      },
+    };
+  }
+
+  async function repairJobs(simulate: boolean = false) {
+    const results: {
+      savedObjectsCreated: string[];
+      savedObjectsDeleted: string[];
+      datafeedsAdded: string[];
+      datafeedsRemoved: string[];
+    } = {
+      savedObjectsCreated: [],
+      savedObjectsDeleted: [],
+      datafeedsAdded: [],
+      datafeedsRemoved: [],
+    };
+
+    const { body: datafeeds } = await client.asInternalUser.ml.getDatafeeds<{
+      datafeeds: Datafeed[];
+    }>();
```

Comment:
I understand we don't need to perform a check in this case, but should we use a custom mlClient for consistency?

Reply:
the datafeeds returned from `mlClient` will be filtered for the space. we need all datafeeds that actually exist for this check